docs/docs/using-pants/key-concepts/options.mdx

```diff
@@ -102,9 +102,16 @@ domain = "my.domain"
 
 [python-repos]
 repo_host = "repo.%(domain)s"
-indexes.add = ["http://%(env.PY_REPO)s@%(repo_host)s/index
+indexes.add = ["https://%(env.PY_REPO)s@%(repo_host)s/index"]
 ```
 
+If you need to set default values for environment variables without requiring all users to define them in the local
+environment, you can export them in the `.pants.bootstrap` Bash script. This file needs to be placed in the root of your
+workspace, and it is going to be sourced before invoking any Pants goal.
+
+You can also add to this file any Bash code you want to execute before Pants runs, and any environment variables
+declared in this file are going to be available to any process that Pants may start.
```

Comment:
I think this is great content, and the first sentence makes sense right here. I wonder if it's worth adding a specific section somewhere about `.pants.bootstrap` which can be linked to from here and from the `ulimit` section, and also from slack/github issues when required for helping someone.

This new section could then have the info you've already written (about how the file works, how one can define env vars and execute other commands), but also have an example of what to do put into it too.

Reply:
(I don't know where this section should be, it doesn't exactly feel like something about `options`)

Reply:
Good idea. I think this section could be renamed from "Initial configuration" to "Pants configuration" to offer a home for general configuration topics, and add a section about this in there.

https://www.pantsbuild.org/2.18/docs/getting-started/initial-configuration

Reply:
Thanks for thinking it through.

That's a critical page for someone getting started with pants, so I'd be a bit nervous about loading it with "advanced" topics that aren't necessary for (most) people to get something useful going, as it will be come overwhelming.

Some other ideas:

1. a new page that's something like "Using the 'pants' launcher" under "Using pants" that describes scie-pants and its features: starting with .pants.bootstrap but could include PANTS_SOURCE etc too. This doesn't feel perfect because a user generally won't care/know what would be a launcher feature vs not a launcher feature (ie if looking for something, wouldn't know to look there to find it)
2. A similar new page framed more as "Customising start-up" or similar

I'm leaning towards 2. Some other ideas? Thoughts?

Reply:
Good points. I took another look at the current options page under Using Pants, and I think having a dedicated section at the bottom, just after the one about the ["`.pants.rc` file"`](https://www.pantsbuild.org/2.18/docs/using-pants/key-concepts/options#pantsrc-file) to introcude the `.pants.bootstrap` file could make sense as a minimal step for now.

Reply:
I like adding another section after ".pants.rc file", will do now.

Reply:
I'd suggest keeping these paragraphs briefer, with a link/mention to the new section for the full explanation.